systemtest/src/test/java/io/strimzi/systemtest/mirrormaker/MirrorMakerIsolatedST.java

```diff
@@ -295,20 +252,13 @@ void testMirrorMakerTlsAuthenticated(ExtensionContext extensionContext) throws E
             .endSpec()
             .build());
 
-        internalKafkaClient = internalKafkaClient.toBuilder()
-            .withTopicName(topicSourceName)
-            .withClusterName(kafkaClusterSourceName)
-            .withKafkaUsername(userSource.getMetadata().getName())
-            .build();
-
-        internalKafkaClient.produceAndConsumesTlsMessagesUntilBothOperationsAreSuccessful();
-
-        internalKafkaClient = internalKafkaClient.toBuilder()
-            .withClusterName(kafkaClusterTargetName)
-            .withKafkaUsername(userTarget.getMetadata().getName())
+        clients = new KafkaClientsBuilder(clients)
+            .withBootstrapAddress(KafkaResources.tlsBootstrapAddress(kafkaClusterTargetName))
+            .withUserName(kafkaTargetUserName)
             .build();
 
-        internalKafkaClient.consumesTlsMessagesUntilOperationIsSuccessful(internalKafkaClient.getMessageCount());
+        resourceManager.createResource(extensionContext, clients.consumerTlsStrimzi(kafkaClusterTargetName));
+        ClientUtils.waitForClientSuccess(testStorage.getConsumerName(), testStorage.getNamespaceName(), MESSAGE_COUNT);
```

Comment:
Wouldn't be better to have constants such as `MESSAGE_COUNT` in `TestStorage` and then re-implement our Utils methods to use only TestStorage as a parameter? Therefore we could avoid parameter order problems..., which unfortunately happens...? (Create an issue for that?)

Reply:
Yeah sure, when I was doing this refactor I had the same idea. I will create an issue for that.

Also, I was thinking about possibility of passing the whole `TestStorage` to the client's builders - in that case we would get rid of the duplicates. WDYT?

Reply:
Yes, remarkable for the simplicity and time saving for the user it's great. (always prefer readability over performance = of course in embedded systems it's not applicable but here we can put the whole `TestStorage` object :) )

Reply:
Created an issue for that #6755 

For the test-clients will be maybe a separate issue - I will create it soon.